eventuate-core/src/test/scala/com/rbmhtechnology/eventuate/EventsourcedActorTestKit.scala

```diff
@@ -0,0 +1,295 @@
+    actor ! Persist(event)
+    expectEvent(event, max)
+  }
+
+  /**
+    * checks if given events were written to eventLog within one [[Write]]. Fails if not.
+    *
+    */
+  def expectEvents[T](expected: Seq[T], max: FiniteDuration = expectEventTimeOut) {
+    processEventlog(Some(expected), max)
+  }
+
+  /**
+    * checks if given events were written to eventLog, each in seperate [[Write]]. Fails if any of given events was
+    * not written, or order of given events differs from persisted events.
+    *
+    */
+  def expectEventsOneAtATime[T](expected: Seq[T], max: FiniteDuration = expectEventTimeOut) {
+    expected foreach { ev =>
+      expectEvent(ev, max)
+    }
+  }
+
+  /** checks if given event was written to eventLog. fails, if not (see expectEvents) */
+  def expectEvent[T](expected: T, max: FiniteDuration = expectEventTimeOut) {
+    processEventlog(Some(Seq(expected)), max)
+  }
+
+
+  // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  // ++++++++++++ cleaning one or multiple events +++++++++++
+  // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+
+  /** Pops one event from event log without returning it */
+  def popFromEventLog(max: FiniteDuration = expectEventTimeOut) {
+    processEventlog(None, max)
+  }
+
+  /**
+    * Pops all events from event log. Can be used to simply clean current event log in order not to explicitly expect
+    * messages that are not of interest and furthermore to 'unblock' actor, which waits for [[WriteSuccess]] messages
+    * before doing anything but persisting again. <br>
+    *
+    * Note: handle with care. In order to be sure that sending events to logProbe is not blocked by waiting for
+    * WriteSuccess in logProbe (which happens if events to be persisted are stashed in persisting actor), logProbe
+    * has to be queried until no more messages can be expected (see processEventlog). Which means this takes at least
+    * max to terminate.
+    */
+  def cleanEventLog(max: FiniteDuration = expectEventTimeOut): Unit = {
+
+    while (Try(processEventlog(None, max)).isInstanceOf[Success[Seq[Any]]]) {
+
+    }
+  }
+
+  // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  // +++++++++++ Querying events currently stashed ++++++++++
+  // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  /** Returns single event currently stashed in event log (which is than popped) */
+  def queryNextEvent(max: FiniteDuration = expectEventTimeOut) = processEventlog(None, max)
+
+  /**
+    * Cleans all events from event log. <br>
+    *
+    * Note: handle with care. In order to be sure that sending events to logProbe is not blocked by waiting for
+    * WriteSuccess in logProbe (which happens if events to be persisted are stashed in persisting actor), logProbe
+    * has to be queried until no more messages can be expected (see processEventlog). Which means this takes at least
+    * max to terminate.
```

Comment:
This should be possible without blocking or waiting `max`. Use a corresponding method on `TestProbe` that returns all received messages.